DuckDuckGo/PassKitPreviewHelper.swift

```diff
@@ -42,3 +44,45 @@ class PassKitPreviewHelper: FilePreview {
         }
     }
 }
+
+class ZippedPassKitPreviewHelper: FilePreview {
+    private weak var viewController: UIViewController?
+    private let filePath: URL
+    
+    required init(_ filePath: URL, viewController: UIViewController) {
+        self.filePath = filePath
+        self.viewController = viewController
+    }
+    
+    func preview() {
+        if let passes: [PKPass] = extractDataFromZip(at: self.filePath)?.compactMap({ try? PKPass(data: $0) }),
+           passes.count > 0,
+           let controller = PKAddPassesViewController(passes: passes) {
+            viewController?.present(controller, animated: true)
+        } else {
+            os_log("Can't present passkit: No passes in passes file", type: .debug)
+        }
+    }
+ 
+    func extractDataFromZip(at zipPath: URL) -> [Data]? {
```

Comment:
This function interface is a bit ambiguous:
-  I would avoid using nil returns as errors. An optional result should be used only in cases where it's really a valid result
- To show failures I would allow the `func` to throw errors, and manage the errors in the caller.
- It's not clear if the `zipPath: URL` can be a local file, a remote URL or both, could you add some standard comments and/or change the param name?